### ISSUE TITLE
how to train own data for multivarite time series forcasting task?

### ISSUE THREAD

**Issue by luoolu** [KindXiaoming/pykan#1]
MLPs already used in time-series forecasting

**Comment by KindXiaoming** [comment]
Interesting application! Both MLPs and KANs are models which cannot do time series immediately. Like one needs to turn MLPs to RNNs/LSTMs/transformers first, we also need to turn KANs into counterparts.

**Comment by luoolu** [comment]
Yes, I understand. I look forward to the application of KAN in this time multivariate sequence prediction task. Thank you very much

**Comment by Somdudewillson** [comment]
Technically, there is the basic method of simply providing a fixed window to the network, which is one I've been experimenting with as part of trying to get a feel for KANs. (Kaggle notebook [here](https://www.kaggle.com/code/somdudewillson/kan-time-series-experiment) if you're curious.)

**Comment by yuedajiong** [comment]
Years ago, I designed a TNN (Temporal Neural Network) based on the concept of Fourier-series. It was intended for time series prediction and anomaly detection, considering that the composition of a time series includes: trend, various periods, and noise. The trend component was calculated using moving averages at different scales, while fitting for multiple periods could be achieved through TNN. The core of TNN involved introducing sin and cos functions as activation functions, with the option to include coefficients to represent amplitude and frequency variations.

**Comment by 2017wxyzwxyz** [comment]
@Somdudewillson,May I ask, based on your previous experience, has KAN significantly improved the prediction accuracy of time series data compared to other models such as LSTM?
